csharp-basic-topics/AwaitInForEachLoops/AwaitInForEachLoops/TaskWhenAll.cs

```diff
@@ -0,0 +1,35 @@
+﻿
+
+namespace AwaitInForEachLoop
+{
+    public class TaskWhenAllInLoop
+    {
+        public static async Task<List<Task>> ResultAsync()
+        {
+            var numbers = new List<int> { 10, 20, 30, 40, 50, 60, 70, 80, 90 };
+
+            var processingTasks = new List<Task>();
+
+            foreach (int number in numbers)
+            {
+            
+                processingTasks.Add(ProcessNumberAsync(number));
+            }
+
+            await Task.WhenAll(processingTasks);
+
+            Console.WriteLine("Done Processing");
+
+            return processingTasks;
```

Comment:
Instead of returning the list of task, return the result. In this case, the order of numbers in the result list may or may not be in the same order as the actual numbers list.

Reply:
Task.WhenAll is a type of Task in c#, so the result we are returning will have to be a type of task. in our case a list of Task.
I have updated the code to look like this
var result = new List<Task>();

foreach (int number in numbers)
{            
    result.Add(ProcessNumberAsync(number));
}